src/Compilers/CSharp/Portable/Errors/LazyMissingNonNullTypesContextDiagnosticInfo.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed class LazyMissingNonNullTypesContextDiagnosticInfo : LazyDiagnosticInfo
+    {
+        private readonly INonNullTypesContext _context;
+        private readonly TypeSymbolWithAnnotations _type;
+
+        internal LazyMissingNonNullTypesContextDiagnosticInfo(INonNullTypesContext context, TypeSymbolWithAnnotations type)
+        {
+            _context = context;
+            _type = type;
+        }
+
+        protected override DiagnosticInfo ResolveInfo()
+        {
+            if (!_type.IsValueType && _context.NonNullTypes != true)
```

Comment:
>if [](start = 12, length = 2)

Can the body of this method be replaced with calling `ReportMissingNonNullTypesContextForAnnotation`?
That will be helpful, as I'm going to pair a language check with the existing check for NonNullTypes context.